packages/coverage-v8/src/browser.ts

```diff
@@ -0,0 +1,69 @@
+import { cdp } from '@vitest/browser/context'
+import type { V8CoverageProvider } from './provider'
+import { loadProvider } from './load-provider'
+
+interface ScriptCoverage {
+  result: Array<{ url: string }>
+}
+
+const session = cdp() as { send: (...options: any[]) => any }
+
+export default {
+  async startCoverage() {
+    await session.send('Profiler.enable')
+    await session.send('Profiler.startPreciseCoverage', {
+      callCount: true,
+      detailed: true,
+    })
+  },
+
+  async takeCoverage() {
+    const coverage: ScriptCoverage = await session.send('Profiler.takePreciseCoverage')
+    const result: typeof coverage.result = []
+
+    // Reduce amount of data sent over rpc by doing some early result filtering
+    for (const entry of coverage.result) {
+      if (filterResult(entry)) {
+        result.push({
+          ...entry,
+          url: decodeURIComponent(entry.url.replace(window.location.origin, '')),
+        })
+      }
+    }
+
+    return { result }
+  },
+
+  async stopCoverage() {
+    await session.send('Profiler.stopPreciseCoverage')
+    await session.send('Profiler.disable')
+  },
+
+  async getProvider(): Promise<V8CoverageProvider> {
+    return loadProvider()
+  },
+}
+
+function filterResult(coverage: ScriptCoverage['result'][number]): boolean {
+  if (!coverage.url.startsWith(window.location.origin)) {
+    return false
+  }
+
+  if (coverage.url.includes('@fs')) {
```

Comment:
Why are these excluded?

Reply:
To exclude files like below:

```
http://localhost:5173/@fs/Users/x/vitest/packages/browser/dist/client.js
http://localhost:5173/@fs/Users/x/vitest/packages/browser/dist/client/__vitest_browser__/preload-helper-Cyh_YkkL.js
```

Reply:
But it's not used only for those things. It's used by Vite for other files. And you already exclude `__vitest_browser__` later. The `browser/dist` will be in `node_modules` so it will also be excluded

Reply:
Alright, removing this should be fine. Though I'd like to add test case that ensures paths with `@fs` are handled correctly when remapping coverage. Any ideas how to create such test case? When would `@fs` appear in source file imports?

Reply:
Files loaded in `test/allow-external.test.ts` seem to trigger this case - perfect! I'll add the required changes and enable that test for browser mode.